### ISSUE TITLE
problem creating unit tests that reference mapbox-gl

### ISSUE THREAD

**Issue by calocan** [mapbox/mapbox-gl-js#4374]
When I try to create a unit test that references mapbox-gl-js, in this case via react-map-gl, I get this error from the window.js object's modification to the jsdom object:

```
 ● Test suite failed to run

    TypeError: Cannot set property clientLeft of [object Object] which has only a getter
      
      at restore (node_modules/mapbox-gl/js/util/window.js:30:45)
      at Object.<anonymous> (node_modules/mapbox-gl/js/util/window.js:65:18)
      at Object.<anonymous> (node_modules/mapbox-gl/js/util/browser.js:8:16)
      at Object.<anonymous> (node_modules/mapbox-gl/js/mapbox-gl.js:3:17)
      at Object.<anonymous> (node_modules/react-map-gl/dist/map.react.js:47:17)
      at Object.<anonymous> (node_modules/react-map-gl/dist/index.js:3:12)
      at Object.<anonymous> (node_modules/react-map-gl/index.js:22:18)
      at Object.<anonymous> (source/components/mapbox/MapBox.js:12:19)
      at Object.<anonymous> (source/components/mapbox/MapBoxContainer.js:14:15)
      at Object.<anonymous> (source/components/mapbox/MapboxContainer.spec.js:19:24)
```

Here is the offending code in window.js:

```
   // Stub some CSSOM-related properties that jsdom doesn't implement.
    window.HTMLElement.prototype.clientLeft = 0;
    window.HTMLElement.prototype.clientTop = 0;

```

I'd like to be able to test React components using Jest and Enzyme, which forces me to test in a node environment instead of the browser. Can you explain why this seemingly erroneous code is called? Is there a better way to test mapbox-gl-js in a node environment? Thanks.

**Comment by calocan** [comment]
It looks like this is solved by installing the same version of jsdom, 9.4.2 as mapbox-gl-js has installed, rather than a newer one. It would be great if there was a section on writing unit tests in applications that use mapbox-gl-js, since some non-obvious problems can occur.

**Comment by mourner** [comment]
I guess we can remove that code if we upgrade jsdom in gl-js — apparently the properties have been implemented.

**Comment by jfirebaugh** [comment]
This was fixed -- #4285 -- it just hasn't made it into a release yet.
